release-notes: rework highlights section (#408325)
NixOS/nixpkgs: nixos/doc/manual/release-notes/rl-2505.section.md

> # NixOS 25.05 (“Warbler”, 2025.05/??) {#sec-release-25.05}

Reviewer: @leona-ya This heading looks incorrect:

> [Release 25.11 (“Xantusia”, 2025.11/??)](file:///nix/store/mhf7cl5y7fvyvqnasrxzkcss2l5barz5-nixos-manual-html/share/doc/nixos/release-notes.html#sec-release-25.11)
> [NixOS 25.05 (“Warbler”, 2025.05/??)](file:///nix/store/mhf7cl5y7fvyvqnasrxzkcss2l5barz5-nixos-manual-html/share/doc/nixos/release-notes.html#sec-release-25.05)
> [Release 24.11 (“Vicuña”, 2024.11/28)](file:///nix/store/mhf7cl5y7fvyvqnasrxzkcss2l5barz5-nixos-manual-html/share/doc/nixos/release-notes.html#sec-release-24.11)

![image](https://github.com/user-attachments/assets/c2c54006-1604-437d-a4f1-c798ff62d320)

Already from commit

https://github.com/NixOS/nixpkgs/commit/63455f1a21b1e5a96104379f01c7436c9af0839a

CC @adfaure @fricklerhandwerk from PR #393738

Author: I'd argue that it is correct and we also want to change this for 25.11. 25.05 is the first release with split release notes, and this change makes this easier to understand in my opinion. But I would like to pass on to @NixOS/documentation-team for a final decision.

## Highlights {#sec-release-25.05-highlights}

<!-- To avoid merge conflicts, consider adding your item at an arbitrary place in the list instead. -->

> In addition to numerous improvements to NixOS modules and general improvemnts, this release has the following highlights:

Reviewer 2: This sounds a little off "numerous improvements to NixOS modules and general improvements" (typo here btw). 

Maybe something like:

```
Alongside many enhancements to NixOS modules and general system improvements, this release features the following highlights:
```

Author: Thank you!

  Refer to the [GNOME release notes](https://release.gnome.org/48/) for more details.

> - [channels.nixos.org](https://channels.nixos.org) now supports the Lockable HTTP Tarball Protocol. This allows to use the channel `nixexprs.tar` as Nix Flake input, e.g.:

Reviewer 2: ```suggestion
- [channels.nixos.org](https://channels.nixos.org) now supports the Lockable HTTP Tarball Protocol. This allows using the channel `nixexprs.tar` as a Nix Flake input, e.g.:
```

Slightly better imho
